sys/net/gnrc/routing/rpl/gnrc_rpl_dodag.c

```diff
@@ -309,9 +309,7 @@ static gnrc_rpl_parent_t *_gnrc_rpl_find_preferred_parent(gnrc_rpl_dodag_t *doda
         return NULL;
     }
 
-    LL_FOREACH(dodag->parents, elt) {
-        new_best = dodag->instance->of->which_parent(new_best, elt);
-    }
+    LL_SORT(dodag->parents, dodag->instance->of->which_parent);
```

Comment:
Yes, but for example with MRHOF as an objective function, the sorting of the parents can vary over time because it depends not only on the advertised rank of a parent, but also on the link quality to the parent.

So sorting the parents on add/del works if the parents have a fixed order, but I don't think it works when the relative sorting of the parents change over time.